### PR TITLE
Sidhant prs

### PR DESCRIPTION
**Pull request by Sidhant-Gupta**
### Have you read the [Contributing Guidelines on Pull Requests]
Yes,I have read.

(Write your answer here.)

### Description

Dutch National Flag Algorithm- Implemented in C++.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] Dutch National Flag algo -Implemented

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

**Comment by Sidhant-Gupta** [comment]
This pr is related to the issue - Dutch National Flag Implementation #1462

**Comment by HarshCasper** [comment]
Hello @Sidhant-Gupta 

If you are picking problems for CP/DS-Algo Portals, please add their link. 

The above problem is from a [LeetCode Problem](https://leetcode.com/problems/sort-colors/).

**Comment by Sidhant-Gupta** [comment]
I've made the desired changes. Please review it.

File: C-Plus-Plus/other/SortColors-DutchNationalFlag.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+// Output: [0,0,1,1,2,2]
+
+// Input:
+// n=3
+// nums = [2,0,1]
+// Output: [0,1,2]
+
+// The bruteforce approach - Sort the vector in nlogn time
+
+//                                 ************************DUTCH_NATIONAL_FLAG ALGORITHM***********************************
```
Comment (HarshCasper) [inline C-Plus-Plus/other/SortColors-DutchNationalFlag.cpp]: Shift the Comments above any lines of code

File: C-Plus-Plus/other/SortColors-DutchNationalFlag.cpp [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+// TEST CASE-
+// Input:
+// n=6
+// nums = [2,0,2,1,1,0]
+// Output: [0,0,1,1,2,2]
+
+// Input:
+// n=3
+// nums = [2,0,1]
+// Output: [0,1,2]
```
Comment (HarshCasper) [inline C-Plus-Plus/other/SortColors-DutchNationalFlag.cpp]: Shift the sample I/O after the Code ends

File: C-Plus-Plus/other/SortColors-DutchNationalFlag.cpp [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                nums[med]=nums[low];
+                nums[low]=temp;
+                low++;
+                med++;
+            }
+            else if(nums[med]==1){
+                med++;
+            }
+            else{
+                // swap(nums[med],nums[high]);
```
Comment (HarshCasper) [inline C-Plus-Plus/other/SortColors-DutchNationalFlag.cpp]: Remove any instances of commented lines of code. We prefer to have comments over the code but not commented code

File: C-Plus-Plus/other/SortColors-DutchNationalFlag.cpp [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      for(int i=0;i<n;i++){
+          cin>>input;
+          nums.push_back(input);
+      }
+      sortColors(nums);
+    }
+
+  //  COMPLEXITY ANALYSIS-
+  //     Time= O(N)
+  //     space=O(1)
```
Comment (HarshCasper) [inline C-Plus-Plus/other/SortColors-DutchNationalFlag.cpp]: Add a line at the EOF

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by Sidhant-Gupta** [comment]
I've made all the requested changes. Kindly review @HarshCasper

**Pull request merged**